### PR TITLE
[kibana] Add pod security policy to template

### PR DESCRIPTION
**Pull request by krichter722**
This is necessary in order to be able to make the chart work on a cluster with pod security policy enabled.

- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [ ] README.md updated with any new values or changes
- [ ] Updated template tests in `${CHART}/tests/*.py` 
- [ ] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

### CODE REVIEW

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
https://github.com/elastic/helm-charts/blob/acb4f5114083a5f26cbec18cdf49b6c11a63db1b/kibana/tests/kibana_test.py#L283

This test is failling in [elastic+helm-charts+pull-request+template-testing/637/CHART=kibana](https://devops-ci.elastic.co/job/elastic+helm-charts+pull-request+template-testing/637/CHART=kibana,label=docker&&virtual/console). Can you fix it?

**Comment by jmlrt** [comment]
jenkins test this please

File: kibana/templates/_helpers.tpl [hunk 1]
```diff
@@ -41,3 +41,16 @@ heritage: {{ .Release.Service }}
 {{ toYaml .Values.labels }}
 {{- end }}
 {{- end -}}
+
+{{/*
+Use the fullname if the serviceAccount value is not set
+*/}}
+{{- define "kibana.serviceAccount" -}}
+{{- if and .Values.rbac.serviceAccountName ( not ( eq .Values.podSecurityPolicy.name "" ) ) -}}
```
Comment (jmlrt) [inline kibana/templates/_helpers.tpl:49]: Why are you checking `podSecurityPolicy.name` value if you're not using `kibana.serviceAccount` in `kibana/templates/podsecuritypolicy.yaml`?

Comment (jmlrt) [reply kibana/templates/_helpers.tpl:49]: I would prefer removing `kibana.serviceAccount` template and keeping `ClusterRole`, `ClusterRoleBinding` and `ServiceAccount` name consistent with what we have in most other charts (you can check `Logstash` chart for that).

File: kibana/values.yaml [hunk 2]
```diff
@@ -72,7 +72,32 @@ securityContext:
...
+    privileged: true
+    fsGroup:
+      rule: RunAsAny
+    runAsUser:
+      rule: RunAsAny
+    seLinux:
+      rule: RunAsAny
+    supplementalGroups:
+      rule: RunAsAny
+    volumes:
```
Comment (jmlrt) [inline kibana/values.yaml:95]: Let's add `downwardAPI` and `emptyDir` like in https://github.com/elastic/helm-charts/pull/496#discussion_r440934948

File: kibana/values.yaml [hunk 3]
```diff
@@ -72,7 +72,32 @@ securityContext:
...
+
+rbac:
+  create: false
+  serviceAccountName: ""
+
+podSecurityPolicy:
+  create: false
+  name: ""
+  spec:
+    privileged: true
```
Comment (jmlrt) [inline kibana/values.yaml:86]: `privileged: true` isn't required for kibana

File: kibana/templates/clusterrole.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: {{ template "kibana.serviceAccount" . }}-cluster-role
+  labels:
+    app: "{{ template "kibana.fullname" . }}"
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+rules:
```
Comment (jmlrt) [inline kibana/templates/clusterrole.yaml:11]: a specific rule for `podsecuritypolicies` resource is required similar to https://github.com/elastic/helm-charts/pull/496#pullrequestreview-431550555

**Comment by krichter722** [comment]
not stale

**Comment by jmlrt** [comment]
closing as stale

**Pull request closed without merging**